packages/dependency-extraction-webpack-plugin/src/index.js

```diff
@@ -0,0 +1,83 @@
+const WPDependencyExtractionWebpackPlugin = require( '@wordpress/dependency-extraction-webpack-plugin' );
+
+const WOOCOMMERCE_NAMESPACE = '@woocommerce/';
+
+/**
+ * Given a string, returns a new string with dash separators converted to
+ * camelCase equivalent. This is not as aggressive as `_.camelCase` in
+ * converting to uppercase, where Lodash will also capitalize letters
+ * following numbers.
+ *
+ * @param {string} string Input dash-delimited string.
+ *
+ * @return {string} Camel-cased string.
+ */
+function camelCaseDash( string ) {
+	return string.replace( /-([a-z])/g, ( _, letter ) => letter.toUpperCase() );
+}
+
+const wooRequestToExternal = ( request ) => {
+	if ( request.startsWith( WOOCOMMERCE_NAMESPACE ) ) {
+		return [
+			'wc',
+			camelCaseDash( request.substring( WOOCOMMERCE_NAMESPACE.length ) ),
+		];
+	}
+};
+
+const wooRequestToHandle = ( request ) => {
+	if ( request.startsWith( WOOCOMMERCE_NAMESPACE ) ) {
+		return 'wc-' + request.substring( WOOCOMMERCE_NAMESPACE.length );
+	}
+};
```

Comment:
This makes the assumption that every import on the `@woocommerce/*` namespace will have an external (and dependency). Woo Blocks uses the `@woocommerce/*` namespace for a number of aliases only. It'd be good if there would be a way we could signal skipping over aliased namespaces via config.